internal/pkg/csumio/csumwriter_test.go

```diff
@@ -97,3 +97,233 @@ func TestWriter_Success(t *testing.T) {
+		name              string
+		description       string
+		preLens           []int
+		targetLen         int
+		expectedChunkLens []int
+	}
+
+	cases := []matrixCase{
+		/* This case is omitted because "capacity > p" is impossible when capacity == chunkSize.
+		{
+			name:        "case01_pGreater_capacityChunk_offsetZero",
+			description: "p > chunk, capacity == chunk (approximates > p), offset zero",
+			preLens:     nil,
+			targetLen:   pGreater,
+			expectedChunkLens: []int{
+				chunkSize,
+				singleByte,
+			},
+		},
+		*/
+		/* this case is omitted because "capacity > p" is impossible when capacity == chunkSize.
+		{
+			name:        "case02_pGreater_capacityChunk_offsetNonZero",
+			description: "p > chunk, capacity == chunk (approximates = p), offset non-zero",
+			preLens: []int{
+				chunkSize,
+			},
+			targetLen: pGreater,
+			expectedChunkLens: []int{
+				chunkSize,
+				chunkSize,
+				singleByte,
+			},
+		},
+		*/
+		{
+			name:        "case03_pGreater_capacityLess_offsetZero",
+			description: "p > chunk, capacity < p, offset zero",
+			preLens: []int{
+				almostFull,
+			},
+			targetLen: pGreater,
+			expectedChunkLens: []int{
+				chunkSize,
+				chunkSize,
+			},
+		},
+		/* this case is omitted because "capacity > p" is impossible when capacity == chunkSize.
+		{
+			name:        "case04_pEqual_capacityChunk_offsetNonZero",
+			description: "p == chunk, capacity == chunk (approximates > p), offset non-zero",
+			preLens: []int{
+				chunkSize,
+			},
+			targetLen: pEqual,
+			expectedChunkLens: []int{
+				chunkSize,
+				chunkSize,
+			},
+		},
+		*/
+		{
+			name:        "case05_pEqual_capacityEqual_offsetZero",
+			description: "p == chunk, capacity == p, offset zero",
+			preLens:     nil,
+			targetLen:   pEqual,
+			expectedChunkLens: []int{
+				chunkSize,
+			},
+		},
+		{
+			name:        "case06_pEqual_capacityLess_offsetNonZero",
+			description: "p == chunk, capacity < p, offset non-zero",
+			preLens: []int{
+				chunkSize,
+				almostFull,
+			},
+			targetLen: pEqual,
+			expectedChunkLens: []int{
+				chunkSize,
+				chunkSize,
+				almostFull,
+			},
+		},
+		{
+			name:        "case07_pSmaller_capacityGreater_offsetZero",
+			description: "p < chunk, capacity > p, offset zero",
+			preLens:     nil,
+			targetLen:   pSmaller,
+			expectedChunkLens: []int{
+				pSmaller,
+			},
+		},
+		{
+			name:        "case08_pSmaller_capacityEqual_offsetNonZero",
+			description: "p < chunk, capacity == p, offset non-zero",
+			preLens: []int{
+				chunkSize,
+				singleByte,
+			},
+			targetLen: pSmaller,
+			expectedChunkLens: []int{
+				chunkSize,
+				chunkSize,
+			},
+		},
+		{
+			name:        "case09_pSmaller_capacityLess_offsetZero",
+			description: "p < chunk, capacity < p, offset zero",
+			preLens: []int{
+				almostFull,
+			},
+			targetLen: pSmaller,
+			expectedChunkLens: []int{
+				chunkSize,
+				twoBytesShort,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
```

Comment:
```suggestion
```

I believe we don't need this since Go 1.22. cf. https://go.dev/blog/loopvar-preview